File: growing-shop-auth/src/main/java/com/example/growingshopauth/company/domain/Company.java
package com.example.growingshopauth.company.domain;

import com.example.growingshopcommon.validator.StringChecker;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class Company {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    @NotBlank
    @Size(max = 30)
    @StringChecker(includeLowerEn = true, includeUpperEn = true, includeNumber = true, includeSpecialCharacter = "-.()[]")
    private String name;

    @Column(nullable = false)
    private String businessRegistrationNumber;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CompanyStatus status;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private CompanyGrade grade = CompanyGrade.BRONZE;
}

File: growing-shop-auth/src/main/kotlin/com/example/growingshopauth/company/domain/Company.kt
package com.example.growingshopauth.company.domain

import com.example.growingshopcommon.entity.BaseEntity
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Pattern
import jakarta.validation.constraints.Size

@Entity
class Company(
    @Column(nullable = false)
    @NotBlank
    @Size(max = 30)
    @Pattern(regexp = "^[a-zA-Z가-힣-.()\\[\\]]+$")
    val name: String,

    @Column(nullable = false)
    val businessRegistrationNumber: String,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val status: CompanyStatus,

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    val grade: CompanyGrade
) : BaseEntity()
